drivers/clock_control/clock_control_nrf_auxpll.c

```diff
@@ -0,0 +1,144 @@
+					     uint32_t *rate)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+	uint8_t range;
+
+	ARG_UNUSED(sys);
+
+	range = AUXPLL_RANGE_FROM_CONFIG(config->auxpll->CONFIG.CFGSTATIC);
+
+	*rate = (range * config->ref_clk_hz +
+		 (config->ref_clk_hz * (uint64_t)config->frequency) / 65536U) /
+		config->out_div;
+
+	return 0;
+}
+
+static enum clock_control_status clock_control_nrf_auxpll_get_status(const struct device *dev,
+								     clock_control_subsys_t sys)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+	uint32_t val;
+
+	ARG_UNUSED(sys);
+
+	val = nrf_auxpll_status_get(config->auxpll) & AUXPLL_STATUS_MODE_Msk;
+	if (val == (AUXPLL_STATUS_MODE_Locked << AUXPLL_STATUS_MODE_Pos)) {
+		return CLOCK_CONTROL_STATUS_ON;
+	}
+
+	return CLOCK_CONTROL_STATUS_OFF;
+}
+
+static struct clock_control_driver_api clock_control_nrf_auxpll_api = {
+	.on = clock_control_nrf_auxpll_on,
+	.off = clock_control_nrf_auxpll_off,
+	.get_rate = clock_control_nrf_auxpll_get_rate,
+	.get_status = clock_control_nrf_auxpll_get_status,
+};
+
+static int clock_control_nrf_auxpll_init(const struct device *dev)
+{
+	const struct clock_control_nrf_auxpll_config *config = dev->config;
+
+	nrf_auxpll_ctrl_frequency_set(config->auxpll, config->frequency);
+
+	nrf_auxpll_lock(config->auxpll);
+	nrf_auxpll_trim_ctune_set(config->auxpll, sys_read8(config->ficr_ctune));
+	config->auxpll->CONFIG.CFGSTATIC = config->cfg;
+	nrf_auxpll_ctrl_outsel_set(config->auxpll, config->out_div);
+	nrf_auxpll_unlock(config->auxpll);
+
+	nrf_auxpll_ctrl_mode_set(config->auxpll, NRF_AUXPLL_CTRL_MODE_LOCKED);
+
+	return 0;
+}
+
+#define CLOCK_CONTROL_NRF_AUXPLL_DEFINE(n)                                                         \
+	static const struct clock_control_nrf_auxpll_config config##n = {                          \
+		.auxpll = (NRF_AUXPLL_Type *)DT_INST_REG_ADDR(n),                                  \
+		.ref_clk_hz = DT_PROP(DT_INST_CLOCKS_CTLR(n), clock_frequency),                    \
+		.ficr_ctune = DT_REG_ADDR(DT_INST_PHANDLE(n, nordic_ficrs)) +                      \
+			      DT_INST_PHA(n, nordic_ficrs, offset),                                \
+		.cfg = (FIELD_PREP(AUXPLL_CONFIG_CFGSTATIC_OUTDRIVE_Msk,                           \
+				   DT_INST_PROP(n, nordic_out_drive)) |                            \
+			FIELD_PREP(AUXPLL_CONFIG_CFGSTATIC_SELCONSTANTI_Msk,                       \
+				   DT_INST_PROP(n, nordic_current_tune)) |                         \
+			FIELD_PREP(AUXPLL_CONFIG_CFGSTATIC_SDMOFF_Msk,                             \
+				   DT_INST_PROP(n, nordic_sdm_disable)) |                          \
+			FIELD_PREP(AUXPLL_CONFIG_CFGSTATIC_SDMDITHEROFF_Msk,                       \
+				   DT_INST_PROP(n, nordic_dither_disable)) |                       \
+			FIELD_PREP(AUXPLL_CONFIG_CFGSTATIC_AUXPLLRANGE_Msk,                        \
+				   DT_INST_ENUM_IDX(n, nordic_range))),                            \
```

Comment:
When using `nrf_auxpll_config_t`, the MDK values are not necessary.

Reply:
As said, this adds unnecessary overhead, so this won't be changed.